### PR TITLE
some (bundled) minor changes

### PR DESCRIPTION
**Pull request by esohns**
Note that this pull request should not simply be merged 'as-is', some commits may need some attention. Others have already been superseded by mainline, but I have found no way to remove them separately through the gihub UI. Specifically:
- 'bd51f4e' has been mostly obsoleted by recent mainline changes (save for changes to ACE/ace/config-macros.h)
- 'd945d56' and '9bc5543' are upstream merge-commits (-- >disregard)

### CODE REVIEW

File: ACE/ace/OS_NS_unistd.inl [hunk 1]
```diff
@@ -1178,10 +1183,12 @@ ACE_OS::unlink (const char *path)
 # elif defined (ACE_LACKS_UNLINK)
   ACE_UNUSED_ARG (path);
   ACE_NOTSUP_RETURN (-1);
-# elif defined (ACE_UNLINK_EQUIVALENT)
-  ACE_OSCALL_RETURN (ACE_UNLINK_EQUIVALENT (path), int, -1);
 # else
+#if defined (_MSC_VER) && (_MSC_VER >= 1800)
+  ACE_OSCALL_RETURN (::_unlink (path), int, -1);
```
Comment (jwillemsen) [inline ACE/ace/OS_NS_unistd.inl]: We have now support for ACE_UNLINK_EQUIVALENT which you removed, this has similar functionality. See also all other ACE_*_EQUIVALENT changes you removed, please correct this in your merge

File: ACE/ace/Task_T.h [hunk 2]
```diff
@@ -1,197 +1,197 @@
...
-#include "ace/Task_T.cpp"
-#endif /* ACE_TEMPLATES_REQUIRE_SOURCE */
-
-#if defined (ACE_TEMPLATES_REQUIRE_PRAGMA)
-#pragma implementation ("Task_T.cpp")
-#endif /* ACE_TEMPLATES_REQUIRE_PRAGMA */
-
-#include /**/ "ace/post.h"
-#endif /* ACE_TASK_T_H */
+// -*- C++ -*-
```
Comment (jwillemsen) [inline ACE/ace/Task_T.h]: Why are these files completely changes according to github? There are more files like this and that makes it really hard to review the real changes, can you see that you make a pull request with only the real changes?

File: ACE/ace/os_include/os_dirent.h [hunk 3]
```diff
@@ -46,7 +46,7 @@ struct dirent {
 };
 #endif /* ACE_HAS_DIRENT */
 
-#if defined (ACE_LACKS_STRUCT_DIR)
+#if defined (ACE_LACKS_DIRENT_H)
```
Comment (jwillemsen) [inline ACE/ace/os_include/os_dirent.h]: This change doesn't look ok, we have a separate define to check if struct dirent is missing, looks better to update ACE_LACKS_STRUCT_DIR to ACE_LACKS_STRUCT_DIRENT

File: ACE/ace/OS_NS_unistd.inl (hunk 1 above)
Comment (esohns) [inline ACE/ace/OS_NS_unistd.inl]: please see my comment on commit bd51f4e

File: ACE/ace/Task_T.h (hunk 2 above)
Comment (esohns) [inline ACE/ace/Task_T.h]: I don't know. These changes are the result of auto-merging upstream. Please disregard commits
'd945d56' and '9bc5543'.

**Comment by jwillemsen** [comment]
With github I can only merge a pull request completely or not, not real easy to select just a set of your changes

File: ACE/ace/os_include/os_dirent.h (hunk 3 above)
Comment (esohns) [inline ACE/ace/os_include/os_dirent.h]: agreed

**Comment by esohns** [comment]
yes, this will be difficult. I will re-integrate the valid changes to a second fork 'ATCD_master' and resubmit

**Comment by jwillemsen** [comment]
Or make a few branches where you separate related changes to one merge request, that is probably easier/safer to review/test

**Pull request closed without merging**